pkg/blobstore/local/periodic_syncer.go

```diff
@@ -144,10 +155,32 @@ func (ps *PeriodicSyncer) ProcessBlockPut() {
 		// The system was idle for some time. Wait a bit, so
 		// that the current epoch gets a meaningful amount of
 		// data.
-		<-ch
+		select {
+		case <-ctx.Done():
+			return false
```

Comment:
This would be the place to do final syncs to storage, right? Right now it just terminates without doing that.

Reply:
Exactly! So that's why my recommendation is to alter PersistentStateSource to have this method:
```go
NotifySyncStarting(isFinalSync bool)
```
And that would cause PersistentBlockList to set the flag.

Reply:
`isFinalSync` has now been added and the (not) potential race of `bl.closedForWriting` has been removed.